### PR TITLE
User Management screen

### LINKED ISSUE #218: Admin User Management Screen

### PR DESCRIPTION
**Pull request by mathieumognol**
Close #218 
Capture d'écran 2024-04-28 094656

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/skysync/screens/userManagement/UserManagementTest.kt [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import ch.epfl.skysync.screens.UserCard
+import ch.epfl.skysync.screens.UserManagementScreen
+import junit.framework.TestCase.assertEquals
+import org.junit.Rule
+import org.junit.Test
+
+class UserManagementTest {
+  @get:Rule val composeTestRule = createComposeRule()
+
+  private val mockUsers =
```
Comment (Prolas) [inline app/src/androidTest/java/ch/epfl/skysync/screens/userManagement/UserManagementTest.kt:24]: You should use DatabaseSetup. There are already users defined

File: app/src/main/java/ch/epfl/skysync/screens/UserManagement.kt [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+          RoleFilter(
+              onRoleSelected = { selectedRole = it },
+              roles = roles,
+          )
+        }
+      },
+      bottomBar = { BottomBar(navController) },
+      floatingActionButton = {
+        FloatingActionButton(
+            onClick = { navController.navigate("addUser") }, containerColor = lightOrange) {
```
Comment (Prolas) [inline app/src/main/java/ch/epfl/skysync/screens/UserManagement.kt:232]: Does this route work ? The route defined in Route seems to be 
```kotlin
const val ADD_USER = "Add User" 
```

**Review by Prolas — COMMENTED** [review]
Overall good, but tiny errors.

File: app/src/main/java/ch/epfl/skysync/screens/UserManagement.kt [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
+@Composable
+fun UserManagementScreen(navController: NavHostController) {
+  var searchQuery by remember { mutableStateOf("") }
+  var selectedRole by remember { mutableStateOf<RoleType?>(null) }
+  val roles = RoleType.entries
+
+  // Filter users based on search query and selected role.
+  val filteredUsers =
+      mockUsers.filter {
```
Comment (Prolas) [inline app/src/main/java/ch/epfl/skysync/screens/UserManagement.kt:209]: Avoid utilizing the hardcoded list directly. Instead, you should pass it as an argument to the UserManagementScreen.

Comment (mathieumognol) [reply app/src/main/java/ch/epfl/skysync/screens/UserManagement.kt]: Yep was for initial preview and is fixed now

File: app/src/main/java/ch/epfl/skysync/screens/UserManagement.kt (hunk 2 above)
Comment (mathieumognol) [reply app/src/main/java/ch/epfl/skysync/screens/UserManagement.kt]: Thanks for noticing it :)

File: app/src/androidTest/java/ch/epfl/skysync/screens/userManagement/UserManagementTest.kt (hunk 1 above)
Comment (mathieumognol) [reply app/src/androidTest/java/ch/epfl/skysync/screens/userManagement/UserManagementTest.kt]: Wasn't aware, is fixed now

**Review by Prolas — APPROVED** [review]
LGTM

**Pull request merged**